src/commands/deploy.rs

```diff
@@ -687,26 +687,31 @@ async fn wait_for_ready(
     }
 }
 
-async fn is_ready(health_check_url: &str) -> bool {
-    let resp = reqwest::get(health_check_url).await;
-    let (msg, ready) = match resp {
-        Err(e) => (format!("error {}", e), false),
-        Ok(r) => {
-            let status = r.status();
-            let ok = status.is_success();
-            let desc = if ok { "ready" } else { "not ready" };
-            (format!("{desc}, code {status}"), ok)
+#[instrument(level = "debug")]
+async fn is_ready(app_info_url: &str, bindle_version: &str) -> Result<bool> {
+    // If the request fails, the app isn't ready
+    let resp = reqwest::get(app_info_url).await?;
+    if !resp.status().is_success() {
+        tracing::debug!("App not ready: {}", resp.status());
+        return Ok(false);
+    }
+    if let Some(active_version) = resp
+        .json::<AppInfo>()
+        .await
+        .ok()
+        .and_then(|info| info.bindle_version)
+    {
+        if active_version != bindle_version {
```

Comment:
Suggest renaming `bindle_version` to `expected_version` to make this check clearer.